chugbot/attendance/roshHome.php

```diff
@@ -0,0 +1,279 @@
+<?php
+    session_start();
+    include_once '../dbConn.php';
+    include_once '../functions.php';
+    bounceToLogin("rosh");
+    checkLogout();
+    setup_camp_specific_terminology_constants();
+
+    $dbErr = "";
+    $sessionId2Name = array();
+    $blockId2Name = array();
+    $groupId2Name = array();
+    $edahId2Name = array();
+    $chugId2Name = array();
+    $bunkId2Name = array();
+
+    fillId2Name(null, $chugId2Name, $dbErr, "chug_id", "chugim", "group_id", "chug_groups");
+    fillId2Name(null, $sessionId2Name, $dbErr, "session_id", "sessions");
+    fillId2Name(null, $blockId2Name, $dbErr, "block_id", "blocks");
+    fillId2Name(null, $groupId2Name, $dbErr, "group_id", "chug_groups");
+    fillId2Name(null, $edahId2Name, $dbErr, "edah_id", "edot");
+    fillId2Name(null, $bunkId2Name, $dbErr, "bunk_id", "bunks");
+
+    echo headerText("Rosh Home");
+
+    if(!attendanceActive()) {
+        $fullErrorMsg = "<div class=\"col-md-6 offset-md-3\"><div class=\"alert alert-danger alert-dismissible fade show m-2\" role=\"alert\">" . 
+        "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button><h5><strong>Error:</strong> " . 
+        "Attendance Disabled</h5>Nothing is available to take attendance for; attendance has been disabled. " .
+        "Contact an administrator if you believe to be receiving this message in error.</div></div>";
+        echo $fullErrorMsg;
+        exit();
+    }
+?>
+    <!-- Uses choices-js, more info at https://github.com/Choices-js/Choices -->
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/choices.js/public/assets/styles/choices.min.css"/>
+    <script src="https://cdn.jsdelivr.net/npm/choices.js/public/assets/scripts/choices.min.js"></script>
+    <script>
+        document.addEventListener('DOMContentLoaded', (event) => {
+
+            const elements = document.querySelectorAll('.choices-js');
+            elements.forEach((element) =>  {
+                const choices = new Choices(element, {
+                    shouldSort: false,
+                    allowHTML: true,
+                    searchChoices: false,
+                    removeItemButton: true
+                });
+            });
+        });
+    </script>
+
+<div class="container row justify-content-center" style="margin:auto;"><div id="errors" class="mt-2"></div></div>
+
+<div class="card card-body mt-2 p-3 mb-3 container">
```

Comment:
Definitely not opposed to breaking this one up, I was trying to figure out a way to make it feel less clunky while still straightforward... maybe have another tab in the "Rosh/Yoetzet" menubar for regular viewer and one for matrix viewer? Or does that become even more confusing to a general user? What are your thoughts on how to restructure this one?

Reply:
It's OK to keep things on one page if they are logically grouped and it simplifies the code. I really like the "card" approach. It might be enough just to add comments before each card that explain what it does, and to separate each card in the code with whitespace. Let me think about this one a little more. :)